rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,849 @@
+      required:
+        - namespace
+      properties:
+        namespace:
+          $ref: '#/components/schemas/Namespace'
+        properties:
+          type: object
+          description: Configured string to string map of properties for the namespace
+          example: '{ "owner": "Hank Bendickson" }'
+          default: '{ }'
+
+    Namespace:
+      description: Reference to one or more levels of a namespace
+      type: array
+      items:
+        type: string
+      example: [ "accounting", "tax" ]
+
+    RenameTableRequest:
+      type: object
+      properties:
+        sourceTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        destinationTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+
+    TableIdentifier:
+      type: object
+      required:
+        - namespace
+        - name
+      properties:
+        namespace:
+          type: array
+          description: Individual levels of the namespace
+          items:
+            type: string
+        name:
+          type: string
+          nullable: false
+
+    UpdateNamespacePropertiesRequest:
+      type: object
+      properties:
+        removals:
+          type: array
+          uniqueItems: true
+          items:
+            type: string
+          example: '[ "department", "access_group" ]'
+        updates:
+          uniqueItems: true
+          type: object
+          items:
+            type: string
+          example: { "owner": "Hank Bendickson" }
+
+
+  #############################
+  # Reusable Response Objects #
+  #############################
+  responses:
+
+    #  Note that this is a representative example response for use as a shorthand in the spec.
+    #  The fields `message` and `type` as indicated here are not presently prescriptive.
+    UnauthorizedResponse:
+      description: Unauthorized. The client does not have permissions to call this route.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/responses/IcebergErrorResponse'
+          example: {
+            "error": {
+              "message": "Not authorized to make this request",
+              "type": "NotAuthorizedException",
+              "code": 401
+            }
+          }
+
+    #  Note that this is a representative example response for use as a shorthand in the spec.
+    #  The fields `message` and `type` as indicated here are not presently prescriptive.
+    UnsupportedOperationResponse:
+      description: Not Acceptable / Unsuported Operation. The server does not support this operation.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/responses/IcebergErrorResponse'
+          example: {
+            "error": {
+              "message": "The server does not support this operation",
+              "type": "UnsupportedOperationException",
+              "code": 406
+            }
+          }
+
+    IcebergErrorResponse:
```

Comment:
if we have this one, why do we need to define the other error responses? My intention is that if we only have a very generic data model defined in open API spec, we don't need to spend effort to keep the error classes in Java and the response here always in sync.

Reply:
The other responses are used to shorten the docs. There are two patterns:
1. Use a common response that has the example and description
2. Use this schema and add custom examples and descriptions

Since 401 never changes, it can use the first pattern. But response codes like 404 use the second pattern because the examples change, like no such table and no such namespace examples.

I think it works to minimize duplication the way @kbendick has it now.

Reply:
thanks for the explanation!